google/cloud/firestore_v1/stream_iterator.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for iterating over stream results for the Google Cloud Firestore API.
+"""
+
+from collections import abc
+
+
+class StreamIterator(abc.Iterator):
+
+    def __init__(self, response_generator):
+        self._generator = response_generator
+
+    def __iter__(self):
+        return self._generator
+    
+    def __next__(self):
+        try:
+            return next(self._generator)
+        except StopIteration:
+            return None
```

Comment:
One thought: previously, we were returning generator objects. If you decide to implement __next__, maybe we should turn this into a full generator and implement __send__ and __throw__ too to be safe?

(I don't think these would be used by current users, but it would keep everything consistent)

Reply:
I feel like I had better implement these, because **send** and **throw** are quite useful in coroutines.

Reply:
If we were to go this route, should we rename the iterator to generator? Are we going too far on this 😂

Reply:
Yeah, I guess since the old type annotation was Generator, maybe we should make this into a full `StreamGenerator` class to be consistent (not that the type annotations can be relied on)